zerver/views/read_indicators.py

```diff
@@ -0,0 +1,29 @@
+from django.http.request import HttpRequest
+from django.http.response import HttpResponse
+
+from zerver.decorator import REQ, has_request_variables
+from zerver.lib.message import access_message
+from zerver.lib.response import json_success
+from zerver.lib.validator import to_non_negative_int
+from zerver.models import UserMessage, UserProfile
+
+
+@has_request_variables
+def read_by_users(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    message_id: int = REQ(converter=to_non_negative_int, path_only=True),
+) -> HttpResponse:
+
+    # Verify user has access to message.
+    access_message(user_profile, message_id)
+
+    user_ids = (
+        UserMessage.objects.filter(message_id=message_id)
+        .extra(
+            where=[UserMessage.where_read()],
+        )
+        .values_list("user_profile_id", flat=True)
+    )
+
+    return json_success({"user_ids": list(user_ids)})
```

Comment:
This looks about right to me!  I think we might need additional filtering to exclude deactivated users; I think that a UX question I don't understand clearly.

Reply:
I'm confused about excluding deactivated users. Since their read activity was before their deactivation, we may have to show their read receipts too. Like we presumably show deactivated users' mentions in messages before their account deactivation.

Reply:
Did this in a different commit to make it easier to remove in case we decide we don't want to filter.